contracts/governance/Vesting/DevelopmentFund.sol

```diff
@@ -121,14 +143,47 @@ contract DevelopmentFund {
 		safeVault = _safeVault;
 		unlockedTokenOwner = _unlockedTokenOwner;
 
-		lastReleaseTime = block.timestamp;
+		lastReleaseTime = _lastReleaseTime;
+		/// If last release time passed is zero, then current time stamp will be used as the last release time.
+		if (_lastReleaseTime == 0) {
+			lastReleaseTime = block.timestamp;
+		}
+
+		/// Checking if the schedule duration and token allocation length matches.
+		require(_releaseDuration.length == _releaseTokenAmount.length, "Release Schedule does not match.");
+
+		/// Finally we update the token release schedule.
+		releaseDuration = _releaseDuration;
+		releaseTokenAmount = _releaseTokenAmount;
+	}
+
+	/**
+	 * @notice This function is called once after deployment for token transfer based on schedule.
```

Comment:
this solution works, though it's not so elegant.
Alternatively you could precompute contract's address before deployment, transfer tokens to this address while it has no  bytecode, and verify inside constructor that the contract has the proper SOV balance. This approach saves some Solidity code, but puts extra stress on your deployment procedure.

Reply:
I don't think I want to stress @jameshowlett977 more than he is already. I think we can take this route.